src/lib/pages/frontsite/OpeningHours/OpeningHours.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+import Overridable from 'react-overridable';
+import { Container, Grid, Header, Icon } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+import { Error } from '@components/Error';
+import { Loader } from '@components/Loader';
+import LocationOpeningHours from './LocationOpeningHours';
+import { sessionManager } from '@authentication/services/SessionManager';
+import _get from 'lodash/get';
+import { AuthenticationGuard } from '@authentication/components/AuthenticationGuard';
+import { Link } from 'react-router-dom';
+import { BackOfficeRoutes } from '@routes/backoffice/backofficeUrls';
+
+class OpeningHours extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchAllLocations = props.fetchAllLocations;
+  }
+
+  componentDidMount() {
+    this.fetchAllLocations();
+  }
+
+  renderItems = () => {
+    const {
+      data: { hits },
+    } = this.props;
+    const locationPid = _get(sessionManager, 'user.locationPid');
+    return hits
```

Comment:
I know it is not going to be a pure functional function, but what about:
```
const newHits = [];
hits.forEach(el => {
    // move the PID location as first in the array
    el.pid === locationPid ?  newHits.unshift(el) : newHits.push(el)
  }
});
```
The sort is nice but it requires some brain to understand it? WDYT?

Reply:
We went for:
```Javascript
_sortBy(hits, h => [h.pid !== locationPid, h.pid])
```